benchmark/benchmark_umf.hpp

```diff
@@ -316,6 +316,140 @@ struct disjoint_pool : public pool_interface<Provider> {
         return "disjoint_pool<" + Provider::name() + ">";
     }
 };
+template <typename Provider>
+struct disjoint_pool_stack : public disjoint_pool<Provider> {
+    using base = disjoint_pool<Provider>;
+
+    std::vector<umf_memory_provider_handle_t> providers;
+    std::vector<umf_memory_pool_handle_t> pools;
+    std::vector<void *> pool_ptrs;
+
+    static constexpr size_t firstPoolSize = 2ull * 1024 * 1024 * 1024; // 2GB
+    static constexpr size_t levels = 7;
+
+    void SetUp(::benchmark::State &state) {
+        base::provider.SetUp(state);
+        if (state.thread_index() != 0) {
+            return;
+        }
+
+        providers.push_back(base::provider.provider);
+        base::provider.provider = NULL;
```

Comment:
If `.provider` is the pointer type, `nullptr` should be used (like 3 lines below). `NULL` can be converted to int as well.

Reply:
done